Calendar/BookingsDiagnosticSummary.ps1

```diff
@@ -0,0 +1,170 @@
+#
+# Get-CalendarDiagnosticObjectsSummary.ps1 -Identity someuser@microsoft.com -Subject "Test OneTime Meeting Subject"
+#
+# Get-CalendarDiagnosticObjectsSummary.ps1 -Identity User1, User2, Delegate -MeetingID $MeetingID
+#
+# Get-CalendarDiagnosticObjectsSummary.ps1 -Identity $Users -MeetingID $MeetingID -TrackingLogs -Exceptions
+#
+param
+(
+    [Parameter(Position=0, Mandatory=$False, HelpMessage="Specifies the Bookings mailbox to be accessed.")]
+    [ValidateNotNullOrEmpty()]
+    [string]$identity,
+
+    [Parameter(Position=1, Mandatory=$False, HelpMessage="Verify Staff permissions for the Bookings mailbox.")]
+    [switch]$Staff,
+
+    [Parameter(Position=2, Mandatory=$False, HelpMessage="Get the Staff Membership Log for the Bookings mailbox.")]
+    [switch]$StaffMembershipLog,
+
+    [Parameter(Position=3, Mandatory=$False, HelpMessage="Use Graph API to get the Bookings mailbox, Staff, Services and Availiability.")]
+    [switch]$Graph,
+
+    [Parameter(Position=4, Mandatory=$False, HelpMessage="Get MessageTrace logs for the Bookings mailbox(Past 5 days).")]
+    [switch]$GetMessageTrace,
+
+    [Parameter(Position=5, Mandatory=$False, HelpMessage="Export all data to Excel.")]
+    [switch]$ExportExcel
+
+)
+
+
+# ===================================================================================================
+# Auto update script
+# ===================================================================================================
+$BuildVersion = ""
+. $PSScriptRoot\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+if (Test-ScriptVersion -AutoUpdate -Confirm:$false) {
+    # Update was downloaded, so stop here.
+    Write-Host "Script was updated. Please rerun the command." -ForegroundColor Yellow
+    return
+}
+
+Write-Verbose "Script Versions: $BuildVersion"
+
+# ===================================================================================================
+# Helper scripts
+# ===================================================================================================
+. $PSScriptRoot\BookingHelpers\BookingGenericFunctions.ps1
+. $PSScriptRoot\BookingHelpers\BookingTenantSettingsHelper.ps1
+. $PSScriptRoot\BookingHelpers\BookingTenantSettingsLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingMBHelpers.ps1
+. $PSScriptRoot\BookingHelpers\BookingMBLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingMessageTrackingLogHelper.ps1
+. $PSScriptRoot\BookingHelpers\BookingMessageTrackingLogLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingStaffHelpers.ps1
+. $PSScriptRoot\BookingHelpers\BookingStaffLogic.ps1
+. $PSScriptRoot\BookingHelpers\bookingStaffLogHelper.ps1
+. $PSScriptRoot\BookingHelpers\bookingStaffLogLogic.ps1
+. $PSScriptRoot\BookingHelpers\fileSaveHelpers.ps1
+. $PSScriptRoot\BookingHelpers\ExcelWrite.ps1
+. $PSScriptRoot\CalLogHelpers\Write-DashLineBoxColor.ps1
+. $PSScriptRoot\CalLogHelpers\ExcelModuleInstaller.ps1
```

Comment:
Move to public Shared
Can be done later, but sooner is better

Reply:
Has to be done later to check impact on CDL script as well